### DISCUSSION TITLE
CoApp not recognized in linuxserver/firefox docker container running on Synology NAS

### DISCUSSION THREAD

**Discussion by ambell7** [aclap-dev/video-downloadhelper#1177] [Help and Q&A]
### Your question

Sorry in advance if the CoApp is simply not supported on this environment but here goes:

Hardware: Synology DS-918+ NAS
NAS O/S: DSM 7.1.1-42962 Update 6
Synology Docker Package Version: 20.10.3-1308
Container Image: linuxserver/firefox:latest 
Firefox Version in container: 128.0.3 (64 bit) for Alpine Linux (Alpine Linux - 1.0)

VdhCoApp install script: 
https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh
BUT I had to comment out the “file $tmpfile” line in the script as the Linux “file” command is not implemented on a Synology NAS

Output from running script on container startup:

Downloading https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh
Extracting tarball…
Registering CoApp
Installing…
VdhCoApp : VdhCoApp is ready to be used
CoApp successfully installed under ‘~/.local/share/vdhcoapp’.

Output from ~/.local/share/vdhcoapp —info

{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/config/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/config",
  "converterBinary": "/config/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

Symptom:
On the More settings display of the Video Download Helper extension in Firefox is says “Companion App not installed” and “No such native application net.downloadhelper.coapp”

Diagnosis to date: 
The contents of the ~/.local/share/vdhcoapp folder look ok but there appears to be no VdhCoApp.json” file in the ~/.mozilla/native-messaging-hosts directory and even when I create one there manually containing the output from the “vdhcoapp —info” command above and then restart Firefox via the “Restart normally…” button on firefox’s “about:profiles” screen this does not fix the problem.

So …
Any help would be gratefully appreciated. It would be great to be able to get this working not just for myself but for annyone else wanting to use this great Firefox extension in this way. Thanks in advance.

### Addon & CoApp version and details

version: 9.1.0.42
target: mozilla
channel: stable
lang: en-US
coapp: {"found":false,"error":"No such native application net.downloadhelper.coapp"}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/Gecko/20100101 Firefox/128.0
{
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 6
}

**Comment by Mattmjs** [comment]
This is out of my depth with this setup you have. What type of app is Firefox running as ?

**Reply by ambell7** [reply]
I have firefox running in a docker container as described here:
https://github.com/linuxserver/docker-firefox

**Reply by Mattmjs** [reply]
I'm a user like you, the dev @paulrouget  would have more insight on this.

**Comment by Verwundert** [comment]
Same here, Qnap and Firefox Docker...

apk add curl
apk add bash
apk add file

Firefox3:/tmp# curl -sSLf https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh | bash
Downloading: https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-linux-x86_64.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 52.8M  100 52.8M    0     0  8307k      0  0:00:06  0:00:06 --:--:-- 9085k
Extracting tarball…
/tmp/vdhcoapp-GjdHjh: bzip2 compressed data, block size = 900k
Registering CoApp
bash: line 76: /root/.local/share/vdhcoapp/vdhcoapp: cannot execute: required file not found
Firefox3:/tmp# 
![Bildschirmfoto 2024-09-09 um 18 32 44](https://github.com/user-attachments/assets/c1222d89-f7d4-456d-bebc-1b8c58ed13b3)





Video Downloader Prime Addon working for me as alternative.

**Reply by CalvinNg-git** [reply]
Same. How to resolve that?

**Comment by krystalcode** [comment]
I had a similar issue running Firefox in a Podman container. I was able to solve this by running Firefox as its own user within the container (i.e. not `root`) mapped to the host user (e.g. `my_user`) so that the wayland display can be shared. This can be achieved by passing `--userns=keep-id:uid=1000,gid=1000`  when starting the pod/container. Replace `1000` with the ID of your user/group on the host.

There's actually more details in my setup, but without having access to the Video DownloadHelper code it's difficult to say exactly which step made it work. I think the above made a difference, but there could be other things as well.
